routes/api/authRoutes.js

```diff
@@ -0,0 +1,150 @@
+const express = require("express");
+const router = express.Router();
+const bcrypt = require("bcrypt");
+const jwt = require("jsonwebtoken");
+const User = require("../../models/user");
+const authenticateToken = require("../../middleware/authenticateToken");
+
+const {
+  userSchema,
+  subscriptionUpdateSchema,
+} = require("../../validation/userSchemas");
+
+require("dotenv").config();
+
+router.post("/register", async (req, res) => {
+  try {
+    const { error, value } = userSchema.validate(req.body);
+    if (error) {
+      return res.status(400).json({ message: error.details[0].message });
+    }
+
+    const existingUser = await User.findOne({ email: value.email });
```

Comment:
Pomyśl w przyszłości żeby zmniejszać zakres try catch bloku, i zwracać bardziej spersonalizowane informacje.

Reply:
Czyli chodziło by o to, by np. zrobić oddzielnie walidacje, poza try...catch, oddzielne bloki dla sprawdzenia czy user istnieje oraz haszowanie hasła i tworzenie usera 

```suggestion
router.post("/register", async (req, res) => {
  // Walidacja 
  const { error, value } = userSchema.validate(req.body);
  if (error) {
    return res.status(400).json({ message: error.details[0].message });
  }

  // Sprawdzenie czy user istnieje
  try {
    const existingUser = await User.findOne({ email: value.email });
    if (existingUser) {
      return res.status(409).json({ message: "Email in use" });
    }
  } catch (error) {
    console.error("Error checking user existence:", error);
    return res.status(500).json({ message: "Error checking user existence" });
  }

  // Haszowanie hasła i tworzenie user'a
  let hashedPassword;
  try {
    hashedPassword = await bcrypt.hash(value.password, 10);
  } catch (error) {
    console.error("Error hashing password:", error);
    return res.status(500).json({ message: "Error hashing password" });
  }

  try {
    const newUser = new User({
      email: value.email,
      password: hashedPassword,
      subscription: "starter",
    });
    await newUser.save();

    res.status(201).json({
      user: {
        email: newUser.email,
        subscription: newUser.subscription,
      },
    });
  } catch (error) {
    console.error("Registration error:", error);
    res.status(500).json({ message: "Internal server error during user registration" });
  }
});
```